### ISSUE TITLE
openssl --version does not return on version 3.0.13 and 3.3.0 sparc v8

### ISSUE THREAD

**Issue by mfe-** [openssl/openssl#24283]
When executing the command `openssl --version' the application print the version but does not exit. It looks like its waiting / hanging.

openssl was build on a sparc v8 using

```
fes-a120d19nas:/backup/fes-readynasduo/openssl-3.0.13# perl configdata.pm --dump
 
Command line (with current working directory = .):

    /c/opt/perl5.10/bin/perl ./Configure

Perl information:

    /c/opt/perl5.10/bin/perl
    5.10.1 for padre-linux

Enabled features:

    aria
    async
    autoalginit
    autoerrinit
    autoload-config
    bf
    blake2
    bulk
    cached-fetch
    camellia
    capieng
    cast
    chacha
    cmac
    cmp
    cms
    comp
    ct
    deprecated
    des
    dgram
    dh
    dsa
    dso
    dtls
    dynamic-engine
    ec
    ec2m
    ecdh
    ecdsa
    engine
    err
    filenames
    gost
    idea
    legacy
    loadereng
    makedepend
    md4
    mdc2
    module
    multiblock
    nextprotoneg
    ocb
    ocsp
    padlockeng
    pic
    pinshared
    poly1305
    posix-io
    psk
    rc2
    rc4
    rdrand
    rfc3779
    rmd160
    scrypt
    secure-memory
    seed
    shared
    siphash
    siv
    sm2
    sm3
    sm4
    sock
    srp
    srtp
    sse2
    ssl
    ssl-trace
    static-engine
    stdio
    tests
    threads
    tls
    ts
    ui-console
    whirlpool
    tls1
    tls1-method
    tls1_1
    tls1_1-method
    tls1_2
    tls1_2-method
    tls1_3
    dtls1
    dtls1-method
    dtls1_2
    dtls1_2-method

Disabled features:

    acvp-tests          [cascade]        OPENSSL_NO_ACVP_TESTS
    afalgeng            [too-old-kernel] OPENSSL_NO_AFALGENG
    asan                [default]        OPENSSL_NO_ASAN
    asm                 [no asm_arch]    OPENSSL_NO_ASM
    buildtest-c++       [default]        
    crypto-mdebug       [default]        OPENSSL_NO_CRYPTO_MDEBUG
    devcryptoeng        [default]        OPENSSL_NO_DEVCRYPTOENG
    ec_nistp_64_gcc_128 [default]        OPENSSL_NO_EC_NISTP_64_GCC_128
    egd                 [default]        OPENSSL_NO_EGD
    external-tests      [default]        OPENSSL_NO_EXTERNAL_TESTS
    fips                [default]        
    fips-securitychecks [cascade]        OPENSSL_NO_FIPS_SECURITYCHECKS
    fuzz-afl            [default]        OPENSSL_NO_FUZZ_AFL
    fuzz-libfuzzer      [default]        OPENSSL_NO_FUZZ_LIBFUZZER
    ktls                [default]        OPENSSL_NO_KTLS
    md2                 [default]        OPENSSL_NO_MD2 (skip crypto/md2)
    msan                [default]        OPENSSL_NO_MSAN
    rc5                 [default]        OPENSSL_NO_RC5 (skip crypto/rc5)
    sctp                [default]        OPENSSL_NO_SCTP
    trace               [default]        OPENSSL_NO_TRACE
    ubsan               [default]        OPENSSL_NO_UBSAN
    unit-test           [default]        OPENSSL_NO_UNIT_TEST
    uplink              [no uplink_arch] OPENSSL_NO_UPLINK
    weak-ssl-ciphers    [default]        OPENSSL_NO_WEAK_SSL_CIPHERS
    zlib                [default]        
    zlib-dynamic        [default]        
    ssl3                [default]        OPENSSL_NO_SSL3
    ssl3-method         [default]        OPENSSL_NO_SSL3_METHOD

Config target attributes:

    AR => "ar",
    ARFLAGS => "qc",
    CC => "gcc",
    CFLAGS => "-Wall -O3",
    CXX => "g++",
    CXXFLAGS => "-Wall -O3",
    HASHBANGPERL => "/usr/bin/env perl",
    RANLIB => "ranlib",
    RC => "windres",
    bn_ops => "BN_LLONG RC4_CHAR",
    build_file => "Makefile",
    build_scheme => [ "unified", "unix" ],
    cflags => "-pthread",
    cppflags => "",
    cxxflags => "-std=c++11 -pthread",
    defines => [ "OPENSSL_BUILDING_OPENSSL" ],
    disable => [  ],
    dso_ldflags => "-Wl,-z,defs",
    dso_scheme => "dlfcn",
    enable => [ "afalgeng" ],
    ex_libs => "-ldl -pthread",
    includes => [  ],
    lflags => "",
    lib_cflags => "",
    lib_cppflags => "-DOPENSSL_USE_NODELETE",
    lib_defines => [  ],
    module_cflags => "-fPIC",
    module_cxxflags => undef,
    module_ldflags => "-Wl,-znodelete -shared -Wl,-Bsymbolic",
    perl_platform => "Unix",
    shared_cflag => "-fPIC",
    shared_defflag => "-Wl,--version-script=",
    shared_defines => [  ],
    shared_ldflag => "-Wl,-znodelete -shared -Wl,-Bsymbolic",
    shared_rcflag => "",
    shared_sonameflag => "-Wl,-soname=",
    shared_target => "linux-shared",
    thread_defines => [  ],
    thread_scheme => "pthreads",
    unistd => "<unistd.h>",

Recorded environment:

    AR = 
    ARFLAGS = 
    AS = 
    ASFLAGS = 
    BUILDFILE = 
    CC = 
    CFLAGS = 
    CPP = 
    CPPDEFINES = 
    CPPFLAGS = 
    CPPINCLUDES = 
    CROSS_COMPILE = 
    CXX = 
    CXXFLAGS = 
    HASHBANGPERL = 
    LD = 
    LDFLAGS = 
    LDLIBS = 
    MT = 
    MTFLAGS = 
    OPENSSL_LOCAL_CONFIG_DIR = 
    PERL = 
    RANLIB = 
    RC = 
    RCFLAGS = 
    RM = 
    WINDRES = 
    __CNF_CFLAGS = 
    __CNF_CPPDEFINES = 
    __CNF_CPPFLAGS = 
    __CNF_CPPINCLUDES = 
    __CNF_CXXFLAGS = 
    __CNF_LDFLAGS = 
    __CNF_LDLIBS = 

Makevars:

    AR              = ar
    ARFLAGS         = qc
    CC              = gcc
    CFLAGS          = -Wall -O3
    CPPDEFINES      = 
    CPPFLAGS        = 
    CPPINCLUDES     = 
    CXX             = g++
    CXXFLAGS        = -Wall -O3
    HASHBANGPERL    = /usr/bin/env perl
    LDFLAGS         = 
    LDLIBS          = 
    PERL            = /c/opt/perl5.10/bin/perl
    RANLIB          = ranlib
    RC              = windres
    RCFLAGS         = 

NOTE: These variables only represent the configuration view.  The build file
template may have processed these variables further, please have a look at the
build file for more exact data:
    Makefile

build file:

    Makefile

build file templates:

    Configurations/common0.tmpl
    Configurations/unix-Makefile.tmpl
```
strace output

```
strace openssl --version
execve("/usr/local/bin/openssl", ["openssl", "--version"], [/* 18 vars */]) = 0
uname({sys="Linux", node="fes-a120d19nas", ...}) = 0
brk(0)                                  = 0x13c000
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/etc/ld.so.preload", O_RDONLY)    = -1 ENOENT (No such file or directory)
open("/opt/gcc-4.6.0/lib/libssl.so.3", O_RDONLY) = -1 ENOENT (No such file or directory)
stat64("/opt/gcc-4.6.0/lib", {st_mode=S_IFDIR|S_ISGID|0755, st_size=16384, ...}) = 0
open("/usr/local/ssl/lib/libssl.so.3", O_RDONLY) = -1 ENOENT (No such file or directory)
stat64("/usr/local/ssl/lib", {st_mode=S_IFDIR|0755, st_size=16384, ...}) = 0
open("/usr/local/lib/libssl.so.3", O_RDONLY) = 3
read(3, "\177ELF\1\2\1\0\0\0\0\0\0\0\0\0\0\3\0\2\0\0\0\1\0\1\272"..., 512) = 512
fstat64(3, {st_mode=S_IFREG|0755, st_size=1621628, ...}) = 0
mmap(NULL, 1480160, PROT_READ|PROT_EXEC, MAP_PRIVATE, 3, 0) = 0x2002c000
mmap(0x2017c000, 114688, PROT_READ|PROT_WRITE|PROT_EXEC, MAP_PRIVATE|MAP_FIXED, 3, 0x140000) = 0x2017c000
close(3)                                = 0
open("/opt/gcc-4.6.0/lib/libcrypto.so.3", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/local/ssl/lib/libcrypto.so.3", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/local/lib/libcrypto.so.3", O_RDONLY) = 3
read(3, "\177ELF\1\2\1\0\0\0\0\0\0\0\0\0\0\3\0\2\0\0\0\1\0\10\206"..., 512) = 512
fstat64(3, {st_mode=S_IFREG|0755, st_size=5992243, ...}) = 0
mmap(NULL, 5322088, PROT_READ|PROT_EXEC, MAP_PRIVATE, 3, 0) = 0x20198000
mprotect(0x20664000, 292200, PROT_NONE) = 0
mmap(0x20668000, 278528, PROT_READ|PROT_WRITE|PROT_EXEC, MAP_PRIVATE|MAP_FIXED, 3, 0x4c0000) = 0x20668000
close(3)                                = 0
open("/opt/gcc-4.6.0/lib/libdl.so.2", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/local/ssl/lib/libdl.so.2", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/local/lib/libdl.so.2", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/etc/ld.so.cache", O_RDONLY)      = 3
fstat64(3, {st_mode=S_IFREG|0644, st_size=34675, ...}) = 0
mmap(NULL, 34675, PROT_READ, MAP_PRIVATE, 3, 0) = 0x2001c000
close(3)                                = 0
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/lib/libdl.so.2", O_RDONLY)       = 3
read(3, "\177ELF\1\2\1\0\0\0\0\0\0\0\0\0\0\3\0\2\0\0\0\1\0\0\36"..., 512) = 512
fstat64(3, {st_mode=S_IFREG|0644, st_size=10512, ...}) = 0
mmap(NULL, 74808, PROT_READ|PROT_EXEC, MAP_PRIVATE, 3, 0) = 0x206ac000
mprotect(0x206b0000, 58424, PROT_NONE)  = 0
mmap(0x206bc000, 16384, PROT_READ|PROT_WRITE|PROT_EXEC, MAP_PRIVATE|MAP_FIXED, 3, 0) = 0x206bc000
close(3)                                = 0
open("/opt/gcc-4.6.0/lib/libpthread.so.0", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/local/ssl/lib/libpthread.so.0", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/local/lib/libpthread.so.0", O_RDONLY) = -1 ENOENT (No such file or directory)
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/lib/libpthread.so.0", O_RDONLY)  = 3
read(3, "\177ELF\1\2\1\0\0\0\0\0\0\0\0\0\0\3\0\2\0\0\0\1\0\0A\314"..., 512) = 512
fstat64(3, {st_mode=S_IFREG|0644, st_size=78150, ...}) = 0
mmap(NULL, 389928, PROT_READ|PROT_EXEC, MAP_PRIVATE, 3, 0) = 0x206c0000
mprotect(0x206cc000, 340776, PROT_NONE) = 0
mmap(0x206d0000, 65536, PROT_READ|PROT_WRITE|PROT_EXEC, MAP_PRIVATE|MAP_FIXED, 3, 0) = 0x206d0000
mmap(0x206e0000, 258856, PROT_READ|PROT_WRITE|PROT_EXEC, MAP_PRIVATE|MAP_FIXED|MAP_ANONYMOUS, -1, 0) = 0x206e0000
close(3)                                = 0
open("/opt/gcc-4.6.0/lib/libc.so.6", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/local/ssl/lib/libc.so.6", O_RDONLY) = -1 ENOENT (No such file or directory)
open("/usr/local/lib/libc.so.6", O_RDONLY) = -1 ENOENT (No such file or directory)
access("/etc/ld.so.nohwcap", F_OK)      = -1 ENOENT (No such file or directory)
open("/lib/libc.so.6", O_RDONLY)        = 3
read(3, "\177ELF\1\2\1\0\0\0\0\0\0\0\0\0\0\3\0\2\0\0\0\1\0\1\317"..., 512) = 512
fstat64(3, {st_mode=S_IFREG|0755, st_size=1303612, ...}) = 0
mmap(NULL, 1373528, PROT_READ|PROT_EXEC, MAP_PRIVATE, 3, 0) = 0x20720000
mprotect(0x20858000, 95576, PROT_NONE)  = 0
mmap(0x20860000, 65536, PROT_READ|PROT_WRITE|PROT_EXEC, MAP_PRIVATE|MAP_FIXED, 3, 0x130000) = 0x20860000
close(3)                                = 0
munmap(0x2001c000, 34675)               = 0
getpid()                                = 19158
rt_sigaction(SIGRT_0, {0x206c7c74, [], 0}, NULL, 0x2075319c, 8) = 0
rt_sigaction(SIGRT_1, {0x206c7ce4, [], 0}, NULL, 0x2075319c, 8) = 0
rt_sigaction(SIGRT_2, {0x206c7e04, [], 0}, NULL, 0x2075319c, 8) = 0
rt_sigprocmask(SIG_BLOCK, [RT_0], NULL, 8) = 0
rt_sigprocmask(SIG_UNBLOCK, [RT_1], NULL, 8) = 0
_sysctl({{CTL_KERN, KERN_VERSION}, 2, 0x3f533800, 31, (nil), 0}) = 0
brk(0)                                  = 0x13c000
brk(0x160000)                           = 0x160000
brk(0)                                  = 0x160000
rt_sigaction(SIGPIPE, {SIG_IGN}, {SIG_DFL}, 0x2075319c, 8) = 0
profil(0x207d57c4, 0x201b7dd4, 0x2043b018, 0x24) = 0
sbrk(545085520)                         = 0
mctl(0x207d58dc, 538672596, 0x2043b03c, 0x24) = 0
sstk(0x207d5968)                        = 0
open("/usr/local/ssl/openssl.cnf", O_RDONLY|O_LARGEFILE) = 3
fstat64(3, {st_mode=S_IFREG|0644, st_size=10835, ...}) = 0
mmap(NULL, 16384, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x2001c000
read(3, "#\n# OpenSSL example configuratio"..., 16384) = 10835
profil(0x23db59c0, 0x20, 0x19740, 0x2086d5bc) = 0
sbrk(0)                                 = 0
mctl(0, 32, 0x19740, 0x2086d5bc)        = 0
sstk(0)                                 = 0
read(3, "", 16384)                      = 0
close(3)                                = 0
munmap(0x2001c000, 16384)               = 0
open("/etc/mtab", O_RDONLY)             = 3
fstat64(3, {st_mode=S_IFREG|0644, st_size=297, ...}) = 0
mmap(NULL, 16384, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x2001c000
read(3, "/dev/hdc1 / ext2 rw,noatime 0 0\n"..., 16384) = 297
close(3)                                = 0
munmap(0x2001c000, 16384)               = 0
open("/proc/meminfo", O_RDONLY)         = 3
fstat64(3, {st_mode=S_IFREG|0444, st_size=0, ...}) = 0
mmap(NULL, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x2001c000
read(3, "MemTotal:      1010656 kB\nMemFre"..., 1024) = 598
close(3)                                = 0
munmap(0x2001c000, 8192)                = 0
profil(0, 0x216d8, 0x1010101, 0x80808080) = 0
sbrk(0)                                 = 0
mctl(0, 136920, MC_SYNC|0x1010100, 0x80808080) = 0
sstk(0)                                 = 0
fstat64(1, {st_mode=S_IFCHR|0600, st_rdev=makedev(136, 1), ...}) = 0
mmap(NULL, 8192, PROT_READ|PROT_WRITE, MAP_PRIVATE|MAP_ANONYMOUS, -1, 0) = 0x2001c000
write(1, "OpenSSL 3.3.0 9 Apr 2024 (Librar"..., 61OpenSSL 3.3.0 9 Apr 2024 (Library: OpenSSL 3.3.0 9 Apr 2024)
) = 61
```

Any help appreciated!

**Comment by nhorman** [comment]
Well, the call it appears you are hanging on starts with a library call to printf, which matches the fd being written to (stdout).  If you're stuck in a write systemcall to stdout, it would suggest to me that stdout is attached to a character device that isn't actually a termainal (or compatible device.  I would start by running the program, letting it hang, identifying the process id and then running:
```
ls -l /proc/<pid>fd/1
```

to see what the fd is pointing to.  That will provide a clue for further investigation

**Comment by paulidale** [comment]
Did you mean `openssl version` ?

**Comment by mfe-** [comment]
Thanks for your tips @nhorman. I did the following:

1.  `ps aux |  grep openssl
root      1200 97.0  0.3 10240 3712 pts/0    R+   08:28   5:29 openssl version
root      1225  0.0  0.1  4096 1216 pts/1    S+   08:34   0:00 grep openssl`
2. `ls -l /proc/1200/fd/1
lrwx------    1 root     root           64 Apr 29 08:39 /proc/1200/fd/1 -> /dev/pts/0`

Another try:

1. `openssl version > output.txt`
2. `ps aux ...`
3. `ls -l /proc/1253/fd/1
l-wx------    1 root     root           64 Apr 29 08:42 /proc/1253/fd/1 -> /root/output.txt`

@paulidale the issue occours with `openssl version` and `openssl --version`

**Comment by nhorman** [comment]
wait, I think I misinterpreted the strace output.  It shows a return value from the write call, so its not stuck there, @paulidale is likely right here, that we're getting into some interactive mode loop.  

the next thing to try would be loading openssl into gdb.  Could you please run the following commands:
```
gdb /path/to/openssl
set args --version
r
```
wait for the hang
then
```
ctrl-c
bt
```

lets see what that stack trace looks like

**Comment by mfe-** [comment]
I executed the above steps, this is the output:

```shell
gdb /opt/openssl-3.3.0/bin/openssl 
GNU gdb 6.3-debian
Copyright 2004 Free Software Foundation, Inc.
GDB is free software, covered by the GNU General Public License, and you are
welcome to change it and/or distribute copies of it under certain conditions.
Type "show copying" to see the conditions.
There is absolutely no warranty for GDB.  Type "show warranty" for details.
This GDB was configured as "sparc-linux"...Using host libthread_db library "/lib/libthread_db.so.1".

(gdb) set args --version
(gdb) r
Starting program: /c/opt/openssl-3.3.0/bin/openssl --version
[Thread debugging using libthread_db enabled]
[New Thread 16384 (LWP 11625)]
OpenSSL 3.3.0 9 Apr 2024 (Library: OpenSSL 3.3.0 9 Apr 2024)

Program received signal SIGINT, Interrupt.
[Switching to Thread 16384 (LWP 11625)]
0x206c9618 in __pthread_acquire () from /lib/libpthread.so.0
(gdb) bt
#0  0x206c9618 in __pthread_acquire () from /lib/libpthread.so.0
#1  0x206c94fc in __pthread_alt_unlock () from /lib/libpthread.so.0
#2  0x206c6b34 in pthread_mutex_unlock () from /lib/libpthread.so.0
#3  0x2043a468 in fallback_atomic_load_n () from /usr/local/lib/libcrypto.so.3
#4  0x204398d0 in ossl_synchronize_rcu () from /usr/local/lib/libcrypto.so.3
#5  0x202f1be8 in CONF_modules_unload () from /usr/local/lib/libcrypto.so.3
#6  0x202f1f88 in ossl_config_modules_free () from /usr/local/lib/libcrypto.so.3
#7  0x20412868 in OPENSSL_cleanup () from /usr/local/lib/libcrypto.so.3
#8  0x20755418 in exit () from /lib/libc.so.6
#9  0x0004f1e4 in main ()
(gdb) 

```

This is a working openssl version

```shell
gdb /usr/local/ssl/bin/openssl 
GNU gdb 6.3-debian
Copyright 2004 Free Software Foundation, Inc.
GDB is free software, covered by the GNU General Public License, and you are
welcome to change it and/or distribute copies of it under certain conditions.
Type "show copying" to see the conditions.
There is absolutely no warranty for GDB.  Type "show warranty" for details.
This GDB was configured as "sparc-linux"...(no debugging symbols found)
Using host libthread_db library "/lib/libthread_db.so.1".

(gdb) set args --version
(gdb) r
Starting program: /c/usr/local/ssl/bin/openssl --version
(no debugging symbols found)
(no debugging symbols found)
(no debugging symbols found)
(no debugging symbols found)
(no debugging symbols found)
openssl:Error: '--version' is an invalid command.

Standard commands
asn1parse         ca                ciphers           cms               
crl               crl2pkcs7         dgst              dh                
dhparam           dsa               dsaparam          ec                
ecparam           enc               engine            errstr            
gendh             gendsa            genpkey           genrsa            
nseq              ocsp              passwd            pkcs12            
pkcs7             pkcs8             pkey              pkeyparam         
pkeyutl           prime             rand              req               
rsa               rsautl            s_client          s_server          
s_time            sess_id           smime             speed             
spkac             srp               ts                verify            
version           x509              

Message Digest commands (see the `dgst' command for more details)
md4               md5               mdc2              rmd160            
sha               sha1              

Cipher commands (see the `enc' command for more details)
aes-128-cbc       aes-128-ecb       aes-192-cbc       aes-192-ecb       
aes-256-cbc       aes-256-ecb       base64            bf                
bf-cbc            bf-cfb            bf-ecb            bf-ofb            
camellia-128-cbc  camellia-128-ecb  camellia-192-cbc  camellia-192-ecb  
camellia-256-cbc  camellia-256-ecb  cast              cast-cbc          
cast5-cbc         cast5-cfb         cast5-ecb         cast5-ofb         
des               des-cbc           des-cfb           des-ecb           
des-ede           des-ede-cbc       des-ede-cfb       des-ede-ofb       
des-ede3          des-ede3-cbc      des-ede3-cfb      des-ede3-ofb      
des-ofb           des3              desx              idea              
idea-cbc          idea-cfb          idea-ecb          idea-ofb          
rc2               rc2-40-cbc        rc2-64-cbc        rc2-cbc           
rc2-cfb           rc2-ecb           rc2-ofb           rc4               
rc4-40            seed              seed-cbc          seed-cfb          
seed-ecb          seed-ofb          


Program exited normally.
(gdb) set args version
(gdb) r
Starting program: /c/usr/local/ssl/bin/openssl version
(no debugging symbols found)
(no debugging symbols found)
(no debugging symbols found)
(no debugging symbols found)
(no debugging symbols found)
OpenSSL 1.0.2r  26 Feb 2019

Program exited normally.
(gdb) quit
```

**Comment by nhorman** [comment]
ah, that explains it.  The introduction of RCU in 3.3.0 ran into an issue on 32 bit systems (to which I think sparc v8 applies).  Atomic loads on platforms that do not use native atomic load/store operations loose information when performing non-native loads/stores.  Its been fixed in a subsequent patch, which will be slated for the 3.3.1 release.  If you're in a hurry you can fix this now by applying commit 8cecc55b852 to your tree (or alternatively just checkout the head of the openssl-3.3 branch from git).  Apologies for the inconvenience .

Please note, sparcv8 is an unadopted platform currently, meaning it doesn't get any testing in our CI infrastructure.  If you plan on using this platform going forward with newer versions of openssl as they are released, I would encourage you to monitor our release [schedule](https://github.com/orgs/openssl/projects/11/views/23), ans as we approach alpha/beta/final release phases, increase your testing, and report any errors that you might encounter so we can endeavor to fix them prior to release.

**Comment by mfe-** [comment]
I downloaded and compiled the openssl version which contains the mentioned fix.

I can verify that the fix is working, see the underneath output. Thank you very much!  

```shell
export LD_LIBRARY_PATH=/opt/openssl-3.3.0/lib/:$LD_LIBRARY_PATH
fes-a120d19nas:~# ldd /opt/openssl-3.3.0/bin/openssl 
	libssl.so.3 => /opt/openssl-3.3.0/lib/libssl.so.3 (0x2002c000)
	libcrypto.so.3 => /opt/openssl-3.3.0/lib/libcrypto.so.3 (0x20120000)
	libdl.so.2 => /lib/libdl.so.2 (0x20490000)
	libpthread.so.0 => /lib/libpthread.so.0 (0x204a4000)
	libc.so.6 => /lib/libc.so.6 (0x20504000)
	/lib/ld-linux.so.2 => /lib/ld-linux.so.2 (0x20000000)
fes-a120d19nas:~# which openssl
/opt/openssl-3.3.0/bin/openssl
fes-a120d19nas:~# openssl version
OpenSSL 3.3.1-dev  (Library: OpenSSL 3.3.1-dev )
uname -a
Linux fes-a120d19nas 2.6.17.14ReadyNAS #1 Wed Jun 20 20:08:20 PDT 2012 padre GNU/Linux
```
